tests/Jellyfin.Server.Integration.Tests/EncodedQueryStringTest.cs

```diff
@@ -0,0 +1,33 @@
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Jellyfin.Server.Integration.Tests
+{
+    /// <summary>
+    /// Defines the test for encoded querystrings in the url.
+    /// </summary>
+    public class EncodedQueryStringTest : IClassFixture<JellyfinApplicationFactory>
+    {
+        private readonly JellyfinApplicationFactory _factory;
+
+        public EncodedQueryStringTest(JellyfinApplicationFactory factory)
```

Comment:
I'm not going to make you change it, but I think you could've used the default `TestServer` for this with a custom controller and registering your middleware. Using the factory is a little overkill.